lib/charms/layer/nagios.py

```diff
@@ -0,0 +1,57 @@
+import os
+import stat
+
+NAGIOS_PLUGINS_DIR = '/usr/lib/nagios/plugins'
+
+
+def install_nagios_plugin(source_path, plugin_name, data=None):
```

Comment:
This API doesn't make sense. `source_path` is a required arg, but it's not used if `data` is passed.

Reply:
This began life as `def install_nagios_plugin(source_path, plugin_name)` and then I realized I needed to template my plugin, which made me then think about writing it to disk and then reading it right back to copy it and that didn't make sense to me. I added data as a way around that. I thought about making source_path either the data or a filename, but I didn't want to rely on filename parsing to guess. Do you have any suggestions here?

Reply:
I think I lean toward:

```python
def install_nagios_plugin(data, plugin_name):
```

I don't think it's too onerous to require the end-user to do:

```python
install_nagios_plugin(Path(source_file).read_text(), plugin_name)
```

Reply:
You could also have:
```python
def install_nagios_plugin_from_text(text, plugin_name):
    # rest of code

def install_nagios_plugin_from_file(source_file_path, plugin_name):
    install_nagios_plugin_from_text(Path(source_file_path).read_text(), plugin_name)
```